Source/UI/IdentityUI.swift

```diff
@@ -477,6 +488,105 @@ extension IdentityUI {
         }
     }
 
+    @available(iOS 13.0, *)
+    private func showSharedWebCredentialsController(client: Client,
+                                                    presentingViewController: UIViewController,
+                                                    localizedTeaserText: String?,
+                                                    scopes: [String],
+                                                    completion: @escaping (Output) -> Void) {
+        func fallback() {
+            show(input: .byLoginMethod(.password,
+                                       presentingViewController: presentingViewController,
+                                       localizedTeaserText: localizedTeaserText,
+                                       scopes: scopes),
+                 client: client,
+                 completion: completion)
+
+            sharedWebCredentialsController = nil
+        }
+
+        sharedWebCredentialsController = SharedWebCredentialsController { [weak self] result in
+            guard case .success((let username, let password)) = result,
+                  let email = EmailAddress(username),
+                  let self = self else {
+                fallback()
+                return
+            }
+
+            self.identityManager.login(username: .email(email),
+                                       password: password,
+                                       scopes: scopes,
+                                       persistUser: true,
+                                       useSharedWebCredentials: false) { result in
+                guard case .success = result else {
+                    fatalError()
+                    return
+                }
+
+                let user = self.identityManager.currentUser
+
+                func success() {
+                    completion(.success(user: user, persistUser: true))
+                }
+
+                func cancel() {
+                    completion(.cancel)
+                }
+
+                func presentTerms() {
+                    self.identityManager.fetchTerms { result in
+                        guard case .success(let terms) = result else {
+                            cancel()
```

Comment:
Does it mean we will be just silently fail if we fail to fetch terms? Shall we logout in this case as well?

Reply:
The existing flow doesn't logout in this case, but it does present a popup with the error. Maybe we should do the same here.

Reply:
This now logs out the user and presents a popup with an error.